packages/appcd-default-plugins/src/index.js

```diff
@@ -189,6 +191,7 @@ export async function installDefaultPlugins(pluginsDir) {
 
 		newWorkspaces.add(`packages/${manifest.name}/${manifest.version}`);
 	}));
+	incompatible.forEach(index => install.splice(index, 1));
```

Comment:
You can't do this. Every time you call `splice()`, you are modifying the array and throwing all your indices off in the `incompatible` array.

Secondly, you don't need the `incompatible` variable since all you need to do is check `newWorkspaces.size` instead of `install.length` on line 198.

Style note, for performance and readability, it is greatly preferred to use `for-of` instead of `forEach()`.

Reply:
Duh, yeah that was totally dumb of me. But what do you mean with just check `newWorkspaces.size`? For what exactly? In my case it's filled with the same entries as `existingWorkspaces` since nothing changed, and which the `!eq(existingWorkspaces, newWorkspaces)` in line 198 already handles. What else would there be to check?

Reply:
I was thinking that you could use the size of `newWorkspaces` as a counter and also got confused by the variable being named `"incompatible"` as it's irrelevant.

How about this?

```
install = (await Promise.all(install.map(async pkg => {
    // the original code
    return pkg;
}))).filter(Boolean);

Reply:
Ah, gotcha!